### ISSUE TITLE
rename_table is broken on PostgreSQL when the schema_search_path contains dots

### ISSUE THREAD

**Issue by frenkel** [rails/rails#21418]
The following will fail when using a schema with dots in it's name, because the quotes are not around the schema name.

config/database.yml:

    adapter: postgresql
    user: example
    database: example
    schema_search_path: '"test.example.com"'

db/migrate/20150827142257_rename_table.rb:
```
class RenameTable < ActiveRecord::Migration
  def change
    rename_table :users, :users_details
  end
end
```

Result:
```
== 20150827142257 RenameTable: migrating ===========================
-- rename_table(:users, :user_details)
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

PG::InvalidSchemaName: ERROR:  schema "gt" does not exist
: ALTER TABLE "test"."example" RENAME TO "user_details_id_seq"
db/migrate/20150827142257_rename_table.rb:10:in `change'
ActiveRecord::StatementInvalid: PG::InvalidSchemaName: ERROR:  schema "test" does not exist
: ALTER TABLE "test"."example" RENAME TO "user_details_id_seq"
db/migrate/20150827142257_rename_table.rb:10:in `change'
PG::InvalidSchemaName: ERROR:  schema "test" does not exist
db/migrate/20150827142257_rename_table.rb:10:in `change'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

After some debugging I found out that `pk_and_sequence_for` will return `'test.example.com'` instead of `'"test.example.com"'`.

**Comment by senny** [comment]
@frenkel thanks for reporting, I'll have a look.

**Comment by frenkel** [comment]
Great, thanks for the fast reply. For now I've switched to schema without dots in their names and that works fine.
Keep up the good work!

**Comment by senny** [comment]
After some preliminary testing I think this problem is not only related to `rename_table` but how we most of the places we use a schema. For example `create_schema` didn't work...

I'll look some more to find out how much of a change this would be.

**Comment by frenkel** [comment]
Didn't know about the `create_schema` method, I've always created them with a query like `CREATE SCHEMA "test.example.com"`.

**Comment by senny** [comment]
I pushed a fix to `master` that fixes `create_schema`, `drop_schema` and `rename_table`.

**Comment by frenkel** [comment]
Great, thank you very much! Will this become available in 4.2.5?

**Comment by senny** [comment]
@frenkel not sure yet. I need to investigate further wether this was the whole story or if we need further adjustments to fully support schemas with ".". If that is the whole patch, a backport seems reasonable.
